packages/@aws-cdk/aws-eks/test/test.nodegroup.ts

```diff
@@ -206,6 +227,185 @@ export = {
     ));
     test.done();
   },
+  'create nodegroup with on-demand capacity type'(test: Test) {
+    // GIVEN
+    const { stack, vpc } = testFixture();
+
+    // WHEN
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      kubectlEnabled: true,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+    });
+    new eks.Nodegroup(stack, 'Nodegroup', {
+      cluster,
+      instanceType: new ec2.InstanceType('m5.large'),
+      capacityType: eks.CapacityType.ON_DEMAND,
+    });
+
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::EKS::Nodegroup', {
+      InstanceTypes: [
+        'm5.large',
+      ],
+      CapacityType: 'ON_DEMAND',
+    },
+    ));
+    test.done();
+  },
+  'create nodegroup with spot capacity type'(test: Test) {
+    // GIVEN
+    const { stack, vpc } = testFixture();
+
+    // WHEN
+    const cluster = new eks.Cluster(stack, 'Cluster', {
+      vpc,
+      kubectlEnabled: true,
+      defaultCapacity: 0,
+      version: CLUSTER_VERSION,
+    });
+    new eks.Nodegroup(stack, 'Nodegroup', {
+      cluster,
+      instanceTypes: [
+        new ec2.InstanceType('m5.large'),
+        new ec2.InstanceType('t3.large'),
+        new ec2.InstanceType('c5.large'),
+      ],
+      capacityType: eks.CapacityType.SPOT,
+    });
+    // THEN
+    expect(stack).to(haveResourceLike('AWS::EKS::Nodegroup', {
+      InstanceTypes: [
+        'm5.large',
+        't3.large',
+        'c5.large',
+      ],
+      CapacityType: 'SPOT',
+    },
+    ));
+    test.done();
+  },
+  'create nodegroup with ondemand capacity type'(test: Test) {
```

Comment:
Isn't this a duplicate of `create nodegroup with on-demand capacity type` on line 230?

Reply:
@iliapolo 

Let me change the name to `create nodegroup with on-demand capacity type and multiple instances`

so we have two test cases:

1.  on-demand capacity type with one instance in `instanceTypes`
2. on-demand capacity type with multiple instances in `instanceTypes`, which is also accepted by CFN. Under the hood the CFN will create an ASG with launch template with multiple instance types but only the first one will be selected. It's not documented specificly in CFN but it's also accepted.

Reply:
@pahud Wait, creating an `on-demand` node group with multiple instance types will result in a launch template with just the first instance type being selected? ignoring all the others?

Reply:
Hold on a few minutes. Let me paste some screenshots here later.

Reply:
If we created the MNG like this


```ts
this.cluster.addNodegroupCapacity('MNG', {
  capacityType: eks.CapacityType.ON_DEMAND,
  desiredSize: 10,
  instanceTypes: [
    new ec2.InstanceType('t3.large'),
    new ec2.InstanceType('m5.large'),
    new ec2.InstanceType('c5.large'),
  ],
})
```

We actually synthesizes to this

```yaml
  ClusterNodegroupMNGF9B82D86:
    Type: AWS::EKS::Nodegroup
    Properties:
      ClusterName:
        Ref: Cluster9EE0221C
      NodeRole:
        Fn::GetAtt:
          - ClusterNodegroupMNGNodeGroupRole3AB840AB
          - Arn
      Subnets:
        - subnet-05b946a306d81274b
        - subnet-0ae2846156ce0118c
        - subnet-063ca7ed4aacdbad7
      AmiType: AL2_x86_64
      ForceUpdateEnabled: true
      InstanceTypes:
        - t3.large
        - m5.large
        - c5.large
      ScalingConfig:
        DesiredSize: 10
        MaxSize: 10
        MinSize: 1
      CapacityType: ON_DEMAND
```

And the MNG will create an ASG with LT with 100% OD + 0% Spot like:

<img width="1075" alt="圖片" src="https://user-images.githubusercontent.com/278432/102682302-80e8af00-4203-11eb-940a-b3f80d816871.png">

<img width="1062" alt="圖片" src="https://user-images.githubusercontent.com/278432/102682265-29e2da00-4203-11eb-9524-164a8ad41abc.png">

<img width="1120" alt="圖片" src="https://user-images.githubusercontent.com/278432/102682257-1d5e8180-4203-11eb-80f8-a48a15519c32.png">

And all the instances will be the first instance type

<img width="1089" alt="圖片" src="https://user-images.githubusercontent.com/278432/102682275-41ba5e00-4203-11eb-9584-9d5f004a2886.png">


I believe this is expected from ASG's perspective when ASG+LT for on-demand instance types, ASG will not span multiple instance types.

Reply:
@pahud Its not that all the other instance types are ignored, the distribution of instance types across the ASG depends on the allocation strategy, which for on-demand is [`prioritized `](https://docs.aws.amazon.com/autoscaling/ec2/userguide/asg-purchase-options.html#asg-allocation-strategies).

To my understanding this means it will try to fulfill all the capacity by trying the first instance type, but **will**  use other types if there is still a capacity shortage and the first type is not depleted.

I kind of lost sight of why we got into this discussion honestly :) - does this make a difference to us somehow?